packages/perseus/src/widgets/interactive-graphs/stateful-mafs-graph.tsx

```diff
@@ -116,12 +116,19 @@ export const StatefulMafsGraph = React.forwardRef<
     const originalPropsRef = useRef(props);
     const latestPropsRef = useLatestRef(props);
     useEffect(() => {
+        if (state.hasBeenInteractedWith) {
+            return;
+        }
         // This conditional prevents the state from being "reinitialized" right
         // after the first render. This is an optimization, but also prevents
         // a bug where the graph would be marked "incorrect" during grading
         // even if the user never interacted with it.
         if (latestPropsRef.current !== originalPropsRef.current) {
-            dispatch(reinitialize(latestPropsRef.current));
+            dispatch(
+                reinitialize({
+                    ...latestPropsRef.current,
+                }),
+            );
```

Comment:
Does this section need to change? I don't think copying the props will have any effect.